src/sysroot.rs

```diff
@@ -98,6 +99,10 @@ version = "0.0.0"
             stoml.push_str(&profile.to_string())
         }
 
+        // rust-src comes with a lockfile for libstd. Use it.
+        let lockfile = src.path().join("..").join("Cargo.lock");
+        fs::copy(lockfile, &td.join("Cargo.lock")).chain_err(|| "couldn't copy lock file")?;
```

Comment:
Should this be a hard error?

Reply:
Why not? All versions of rust-src that we support have this file. It is present [since 2017](https://github.com/rust-lang/rust/pull/44076).

Reply:
What if you're getting your source not from the `rust-src` package? Like git?

Reply:
Then the lock file [exists in the same place](https://github.com/rust-lang/rust/blob/master/Cargo.lock).

Reply:
Sure but the user may have deleted it. I'm ok with forcing that it exists, but the error message should be slightly clearer.

Reply:
Fair. I made it "Cargo.lock file is missing from source dir".